be/src/vec/columns/predicate_column.h

```diff
@@ -154,8 +153,16 @@ class PredicateColumnType final : public COWHelper<IColumn, PredicateColumnType<
 
     size_t size() const override { return data.size(); }
 
-    [[noreturn]] StringRef get_data_at(size_t n) const override {
-        LOG(FATAL) << "get_data_at not supported in PredicateColumnType";
+    StringRef get_data_at(size_t n) const override {
+        if constexpr (std::is_same_v<T, StringRef>) {
+            auto res = reinterpret_cast<const StringRef&>(data[n]);
+            if constexpr (Type == TYPE_CHAR) {
+                res.size = strnlen(res.data, res.size);
```

Comment:
strnlen may be less then CHAR schema length.

Reply:
it has to be trimed \0